torch/csrc/distributed/c10d/init.cpp

```diff
@@ -946,6 +946,9 @@ Inserts the key-value pair into the store based on the supplied ``key`` and
 ``value``. If ``key`` already exists in the store, it will overwrite the old
 value with the new supplied ``value``.
 
+Note: if the store is wrapped with PrefixStore, a prefix will be added to
```

Comment:
`PrefixStore` allows you to access the underlying store:

https://github.com/pytorch/pytorch/blob/5866284d4a531e47c63ca72f4ad7a8c43a96f789/torch/csrc/distributed/c10d/init.cpp#L1428-L1431

Wouldn't it make more sense to recursively call this and then call `set` on that base store rather than introduce a new API?

Alternatively, since there are multiple levels of prefix stores, we can introduce a new API on prefix store like `base_store` which get the base store and then perform the op there.

I think the new API may be a bit confusing since `TCPStore` is not supposed to understand a concept of a prefix but now we are introducing that logic.

Reply:
Actually this makes more sense after thinking about this more. Because it's always TCPStore, FileStore or HashStore in the innermost and then we wrap with PrefixStore. And the order is fixed because we cannot use a store to initialize a TCPStore or FileStore and we can only use an existing store to initialize a prefixStore.. Maybe this is the cleanest way to solve the issue we encounter. cc: @wconstab

Reply:
One down side here is that, folks wrap PrefixStore with a python class, we might also need to add a similar function in user code as well.